include/af/sparse.h

```diff
@@ -0,0 +1,70 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+#include <af/defines.h>
+
+typedef void * af_sparse_array;
+
+#ifdef __cplusplus
+namespace af
+{
+    class array;
+
+}
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    AFAPI af_err af_create_sparse_array(
+                 af_sparse_array *out,
+                 const dim_t nRows, const dim_t nCols, const dim_t nNZ,
+                 const af_array values, const af_array rowIdx, const af_array colIdx,
+                 const af_sparse_storage storage);
+
+    AFAPI af_err af_create_sparse_array_from_host(
+                 af_sparse_array *out,
+                 const dim_t nRows, const dim_t nCols, const dim_t nNZ,
+                 const void * const values,
+                 const int * const rowIdx, const int * const colIdx,
+                 const af_dtype type, const af_sparse_storage storage);
+
+    AFAPI af_err af_create_sparse_array_from_dense(
+                 af_sparse_array *out, const af_array in,
+                 const af_sparse_storage storage);
+
+    AFAPI af_err af_retain_sparse_array(af_sparse_array *out, const af_sparse_array in);
+
+    AFAPI af_err af_get_sparse_values(af_array *out, const af_sparse_array in);
```

Comment:
I actually prefer these functions being `af_sparse_get*`. Since the class for this would be af::sparse.